### PR TITLE
[12.0][MIG] account_reporting_weight: Migration to v12.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT20199
@chienandalu @Tardo Review please!!

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Tested on runbot :+1:

File: account_reporting_weight/reports/account_invoice_report.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import odoo.addons.decimal_precision as dp
+
+
+class AccountInvoiceReport(models.Model):
+    _inherit = "account.invoice.report"
+
+    weight = fields.Float(digits=dp.get_precision('Stock Weight'))
+
+    def _select(self):
+        select_str = super(AccountInvoiceReport, self)._select()
```
Comment (Tardo) [inline account_reporting_weight/reports/account_invoice_report.py]:
```suggestion
        select_str = super()._select()
```

File: account_reporting_weight/reports/account_invoice_report.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        select_str += """
+            , sub.weight as weight
+            """
+        return select_str
+
+    def _sub_select(self):
+        select_str = super(AccountInvoiceReport, self)._sub_select()
```
Comment (Tardo) [inline account_reporting_weight/reports/account_invoice_report.py]:
```suggestion
        select_str = super()._sub_select()
```

File: account_reporting_weight/reports/account_invoice_report.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        select_str += """
+            , SUM(pr.weight * ail.quantity / u.factor * u2.factor)
+            AS weight
+            """
+        return select_str
+
+    def _group_by(self):
+        group_by_str = super(AccountInvoiceReport, self)._group_by()
```
Comment (Tardo) [inline account_reporting_weight/reports/account_invoice_report.py]:
```suggestion
        group_by_str = super()._group_by()
```

**Review by Tardo — APPROVED** [review]
:+1: LGTM, tested on runbot.

Only a few suggestions to be more python3, and you can squash the bots commits

**Comment by sergio-teruel** [comment]
Ok @Tardo changes done!!

**Pull request merged**